src/virtual_dom/mod.rs

```diff
@@ -39,8 +39,8 @@ type Listeners<COMP> = Vec<Box<dyn Listener<COMP>>>;
 /// A map of attributes.
 type Attributes = HashMap<String, String>;
 
-/// A set of classes.
-type Classes = HashSet<String>;
+/// A list of classes.
+type Classes = Vec<String>;
```

Comment:
I think we could get the best of both worlds (fast difference calculation, and order preservation) with this crate: https://docs.rs/indexmap/1.0.2/indexmap/index.html